src/web/components/Hydrate.tsx

```diff
@@ -1,22 +1,31 @@
+import { useState } from 'react';
 import ReactDOM from 'react-dom';
 
+import { initPerf } from '@root/src/web/browser/initPerf';
+
 type Props = {
     root: IslandType;
-    index?: number;
     children: JSX.Element;
+    index?: number;
+    waitFor?: unknown[];
+};
+
+const isReady = (dependencies: unknown[]): boolean => {
+    return dependencies.every((dep) => dep !== undefined);
 };
 
-export const Hydrate = ({ root, index, children }: Props) => {
-    const rootWithIndex = index === 0 || index ? `${root}-${index}` : root;
-    const element = document.getElementById(rootWithIndex);
+export const Hydrate = ({ root, children, index, waitFor = [] }: Props) => {
```

Comment:
`HydrateOnce`? Might make it clearer and align with `useOnce`

Reply:
Done